webcomponents/tool/src/main/frontend/packages/sakai-lti-popup/src/SakaiLTIPopup.js

```diff
@@ -29,21 +30,22 @@ export class SakaiLTIPopup extends SakaiElement {
   }
 
   launchPopup() {
-
     window.open(this.launchUrl, "_blank");
     document.getElementById("sakai-lti-popup-" + this.randomId).style.display = "none";
     document.getElementById("sakai-lti-popup-hidden-" + this.randomId).style.display = "block";
     return false;
   }
 
-  shouldUpdate() {
-    return this.preLaunchText && this.postLaunchText && this.launchUrl;
+  shouldUpdate(changedProperties) {
+    return changedProperties.has("preLaunchText") || changedProperties.has("postLaunchText") ||
+          changedProperties.has("launchUrl") || changedProperties.has("auto");
   }
 
-  firstUpdated(changedProperties) {
-    changedProperties.forEach((oldValue, propName) => {
-      if ( propName == "auto" && this.auto ) setTimeout(this.launchPopup(), 1000);
-    });
+  firstUpdated() {
+    if ( this.auto && ! this.popped ) {
+      this.popped = true;
```

Comment:
I'm pretty sure that if you make this reactive and just use this.popped in your template, you wont need this timeout at all. Also, you don't need the randomid if you use this.querySelector because it's scoped for you already.